.NET/Microsoft.Recognizers.Text.Sequence/Extractors/BaseSequenceExtractor.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Recognizers.Text.Sequence
+{
+    public abstract class BaseSequenceExtractor : IExtractor
+    {
+        internal abstract ImmutableDictionary<Regex, string> Regexes { get; }
+
+        protected virtual string ExtractType { get; } = "";
+
+        public List<ExtractResult> Extract(string text)
```

Comment:
split big methods to small methods

Reply:
I think the logic here is hard to split, but I have added some comments. Hope this do help.